path_config.go

```diff
@@ -0,0 +1,198 @@
+			logical.ReadOperation:   b.pathConfigRead,
+			logical.UpdateOperation: b.pathConfigWrite,
+		},
+
+		HelpSynopsis:    confHelpSyn,
+		HelpDescription: confHelpDesc,
+	}
+}
+
+func (b *jwtAuthBackend) config(ctx context.Context, s logical.Storage) (*jwtConfig, error) {
+	b.l.RLock()
+	defer b.l.RUnlock()
+
+	if b.cachedConfig != nil {
+		return b.cachedConfig, nil
+	}
+
+	entry, err := s.Get(ctx, configPath)
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	result := &jwtConfig{}
+	if entry != nil {
+		if err := entry.DecodeJSON(result); err != nil {
+			return nil, err
+		}
+	}
+
+	for _, v := range result.JWTValidationPubKeys {
+		key, err := certutil.ParsePublicKeyPEM([]byte(v))
+		if err != nil {
+			return nil, errwrap.Wrapf("error parsing public key: {{err}}", err)
+		}
+		result.ParsedJWTPubKeys = append(result.ParsedJWTPubKeys, key)
+	}
+
+	b.cachedConfig = result
+
+	return result, nil
+}
+
+func (b *jwtAuthBackend) pathConfigRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	config, err := b.config(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if config == nil {
+		return nil, nil
+	}
+
+	resp := &logical.Response{
+		Data: map[string]interface{}{
+			"oidc_issuer_url":        config.OIDCIssuerURL,
+			"oidc_issuer_ca_pem":     config.OIDCIssuerCAPEM,
+			"jwt_validation_pubkeys": config.JWTValidationPubKeys,
+			"bound_issuer":           config.BoundIssuer,
+		},
+	}
+
+	return resp, nil
+}
+
+func (b *jwtAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	config := &jwtConfig{
+		OIDCIssuerURL:        d.Get("oidc_issuer_url").(string),
+		OIDCIssuerCAPEM:      d.Get("oidc_issuer_ca_pem").(string),
+		JWTValidationPubKeys: d.Get("jwt_validation_pubkeys").([]string),
+		BoundIssuer:          d.Get("bound_issuer").(string),
+	}
+
+	// Run checks on values
+	switch {
+	case config.OIDCIssuerURL == "" && len(config.JWTValidationPubKeys) == 0,
+		config.OIDCIssuerURL != "" && len(config.JWTValidationPubKeys) != 0:
+		return logical.ErrorResponse("exactly one of 'oidc_issuer_url' and 'jwt_validation_pubkeys' must be set"), nil
+
+	case config.OIDCIssuerURL != "":
+		_, err := b.createProvider(ctx, config)
+		if err != nil {
+			return logical.ErrorResponse(errwrap.Wrapf("error checking issuer URL: {{err}}", err).Error()), nil
+		}
+
+	case len(config.JWTValidationPubKeys) != 0:
+		for _, v := range config.JWTValidationPubKeys {
+			if _, err := certutil.ParsePublicKeyPEM([]byte(v)); err != nil {
+				return logical.ErrorResponse(errwrap.Wrapf("error parsing public key: {{err}}", err).Error()), nil
+			}
+		}
+
+	default:
+		return nil, errors.New("unknown condition")
+	}
+
+	entry, err := logical.StorageEntryJSON(configPath, config)
+	if err != nil {
+		return nil, err
+	}
+	if err := req.Storage.Put(ctx, entry); err != nil {
+		return nil, err
+	}
+
```

Comment:
Can you call `b.reset()` down here to invalidate the existing config?

Reply:
Yep, good catch.